zsign.cpp

```diff
@@ -64,6 +65,31 @@ int usage()
 	return -1;
 }
 
+string getCacheDirectory() {
+    try {
+#ifdef _WIN32
+        const char* buffer = std::GetEnvironmentVariable("LOCALAPPDATA", nullptr, 0);
+		string cacheDir = std::filesystem::path(buffer) / "zsign_cache";
+        if (!buffer) {
+            throw std::runtime_error("LOCALAPPDATA environment variable is not set.");
+        }
```

Comment:
Likely, you can put this check + `throw` just above line 72, i.e. first check for non-null `buffer`, then construct the final path in `cacheDir`.